suite-common/message-system/src/messageSystemSelectors.ts

```diff
@@ -16,53 +19,57 @@ export const selectMessageSystemCurrentSequence = (state: MessageSystemRootState
-            .map(action => action.message);
-
-        if (!messages?.length) return [];
-
-        return messages.sort(comparePriority);
-    });
+    createAppSelector(
+        [
+            state => state.messageSystem.config,
+            state => state.messageSystem.validMessages[category],
+            state => state.messageSystem.dismissedMessages,
+        ],
+        (config, validMessages, dismissedMessages) => {
+            const nonDismissedValidMessages = validMessages.filter(
+                id => !dismissedMessages[id]?.[category],
+            );
+
+            const messages = config?.actions
+                .filter(({ message }) => nonDismissedValidMessages.includes(message.id))
+                .map(action => action.message);
+
+            return returnStableArrayIfEmpty(messages?.sort(comparePriority));
+        },
+    );
 
 export const selectActiveBannerMessages = makeSelectActiveMessagesByCategory('banner');
 export const selectActiveContextMessages = makeSelectActiveMessagesByCategory('context');
 export const selectActiveModalMessages = makeSelectActiveMessagesByCategory('modal');
 export const selectActiveFeatureMessages = makeSelectActiveMessagesByCategory('feature');
 
-export const selectIsAnyBannerMessageActive = (state: MessageSystemRootState) => {
-    const activeBannerMessages = selectActiveBannerMessages(state);
-
-    return activeBannerMessages.length > 0;
-};
-
-export const selectBannerMessage = memoize((state: MessageSystemRootState) => {
-    const activeBannerMessages = selectActiveBannerMessages(state);
-
-    return activeBannerMessages[0];
-});
+export const selectIsAnyBannerMessageActive = createAppSelector(
+    [selectActiveBannerMessages],
+    activeBannerMessages => activeBannerMessages.length > 0,
+);
 
-export const selectContextMessage = memoizeWithArgs(
-    (state: MessageSystemRootState, domain: ContextDomain) => {
-        const activeContextMessages = selectActiveContextMessages(state);
+export const selectBannerMessage = createAppSelector(
+    [selectActiveBannerMessages],
+    activeBannerMessages => activeBannerMessages[0],
+);
 
-        return activeContextMessages.find(message => message.context?.domain === domain);
-    },
+export const selectContextMessage = createAppSelector(
+    [selectActiveContextMessages, (_state, domain: ContextDomain) => domain],
+    (activeContextMessages, domain) =>
+        activeContextMessages.find(message => message.context?.domain === domain),
 );
 
-export const selectContextMessageContent = memoizeWithArgs(
-    (state: MessageSystemRootState, domain: ContextDomain, language: string) => {
-        const activeContextMessages = selectActiveContextMessages(state);
+export const selectContextMessageContent = createAppSelector(
+    [
+        selectActiveContextMessages,
+        (_state, domain: ContextDomain) => domain,
+        (_state, _domain, language: string) => language,
```

Comment:
nit: 
```suggestion
        (_state, domain: ContextDomain, language: string) => {domain, language},
```

Reply:
I am not sure, but I think this will break memoization, but IDK